runner/utils/driver/docker/docker.go

```diff
@@ -570,6 +592,91 @@ func (d *Docker) GetServices(ctx context.Context, projectID string) ([]*model.Se
 	return serviceArr, nil
 }
 
+// GetServiceStatus gets the status of service info from docker container
+func (d *Docker) GetServiceStatus(ctx context.Context, projectID string) ([]*model.ServiceStatus, error) {
+	networkArgs := filters.Arg("label", "app=service")
+	args := filters.Arg("name", getCurrentProjectServicesName(d.clusterName, projectID))
+	containers, err := d.client.ContainerList(ctx, types.ContainerListOptions{Filters: filters.NewArgs(networkArgs, args), All: true})
+	if err != nil {
+		logrus.Errorf("error getting service in docker unable to list containers got error message - %v", err)
+		return nil, err
+	}
+
+	serviceMapper := make(map[string]map[string][]string)
```

Comment:
I would suggest you have a `map[string][]string`. The key would be `serviceID--version`.